sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/reflect/DoFnSignaturesSplittableDoFnTest.java

```diff
@@ -956,6 +983,96 @@ public void splitRestriction(
     DoFnSignatures.getSignature(BadFn.class);
   }
 
+  @Test
+  public void testTruncateRestrictionReturnsWrongType() throws Exception {
```

Comment:
nit: rename this test and the other ones like this since we aren't returning but having the wrong output receiver type
```suggestion
  public void testTruncateRestrictionWithOutputReceiverWithWrongGenericType() throws Exception {
```